controllers/spacebindingcleanup/spacebinding_cleanup_controller_test.go

```diff
@@ -0,0 +1,127 @@
+package spacebindingcleanup
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/test/space"
+	sb "github.com/codeready-toolchain/host-operator/test/spacebinding"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/masteruserrecord"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestDeleteSpaceBinding(t *testing.T) {
+	// given
+	sbLaraRedhatAdmin := sb.NewSpaceBinding("lara-admin", "lara", "redhat", "admin")
+	sbJoeRedhatView := sb.NewSpaceBinding("joe-view", "joe", "redhat", "view")
+	sbLaraIbmEdit := sb.NewSpaceBinding("lara-edit", "lara", "ibm", "edit")
+
+	redhatSpace := space.NewSpace("redhat")
+	ibmSpace := space.NewSpace("ibm")
+
+	laraMur := masteruserrecord.NewMasterUserRecord(t, "lara")
+	joeMur := masteruserrecord.NewMasterUserRecord(t, "joe")
+
+	t.Run("lara-admin SpaceBinding removed when redhat space is missing", func(t *testing.T) {
+
+		reconciler, request, fakeClient := prepareReconciler(t, sbLaraRedhatAdmin, sbJoeRedhatView, sbLaraIbmEdit, laraMur, joeMur, ibmSpace)
+
+		// when
+		_, err := reconciler.Reconcile(context.TODO(), request)
+
+		// then
+		require.NoError(t, err)
+		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara-admin", fakeClient).DoesNotExist()
+		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "joe-view", fakeClient).Exists()
+		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara-edit", fakeClient).Exists()
+	})
+
+	t.Run("joe-view SpaceBinding removed when joe MUR space is missing", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("joe-view SpaceBinding removed when joe MUR is missing", func(t *testing.T) {
```

Reply:
this particular test verifies that SpaceBinding is deleted when the MasterUserRecord `joe` is missing. The `redhat` space is present

Reply:
I'm confused. I agree with what you said and `joeMur` is the only thing missing from the `prepareReconcile` so shouldn't the test description say `when joe MUR is missing` instead of `when joe MUR space is missing`? What is a `MUR space`? 🤔

Reply:
sorry - this is one big :man_facepalming: 
I somehow thought that you wanted to add the `space` word, and not that you wanted remove it :man_facepalming: 
no idea what I was looking at :man_shrugging: :sweat_smile:

Reply:
removed 2f4f68b6